Source/Core/DolphinQt/SkylanderPortal/SkylanderPortalWindow.cpp

```diff
@@ -92,71 +181,292 @@ void SkylanderPortalWindow::CreateMainWindow()
+  m_element_filter[4]->setText(tr("Fire"));
+  m_element_filter[5]->setText(tr("Earth"));
+  m_element_filter[6]->setText(tr("Life"));
+  m_element_filter[7]->setText(tr("Air"));
+  m_element_filter[8]->setText(tr("Undead"));
+  m_element_filter[9]->setText(tr("Other"));
+
+  search_radio_group->setLayout(search_radio_layout);
+  search_radio_scroll_area->setWidget(search_radio_group);
+  search_element_layout->addWidget(search_radio_scroll_area);
+  search_element_group->setLayout(search_element_layout);
+  search_element_group->setSizePolicy(QSizePolicy::Maximum, QSizePolicy::Preferred);
+  search_filters_layout->addWidget(search_element_group);
+
+  // Widget: Other Filters
+  auto* other_box = new QGroupBox(tr("Other"));
+  auto* other_layout = new QVBoxLayout;
+  connect(m_only_show_collection, &QCheckBox::toggled, this, &SkylanderPortalWindow::RefreshList);
+  other_layout->addWidget(m_only_show_collection);
+  other_box->setLayout(other_layout);
+  search_filters_layout->addWidget(other_box);
+
+  search_checkbox_scroll_area->setMinimumWidth(search_radio_group->width() + 20);
+  search_radio_scroll_area->setMinimumWidth(search_radio_group->width() + 20);
+  other_box->setMinimumWidth(search_radio_group->width() + 20);
+
+  search_filters_layout->addStretch(1);
+
+  finder_list_layout->addLayout(search_filters_layout);
+
+  finder_list_layout->addWidget(m_skylander_list);
+
+  finder_layout->addLayout(finder_list_layout);
+
+  main_layout->addLayout(finder_layout);
+
+  return main_layout;
 }
 
-CreateSkylanderDialog::CreateSkylanderDialog(QWidget* parent) : QDialog(parent)
+bool SkylanderPortalWindow::eventFilter(QObject* object, QEvent* event)
 {
-  setWindowTitle(tr("Skylander Creator"));
-  setObjectName(QString::fromStdString("skylanders_creator"));
-  setMinimumSize(QSize(500, 150));
-  auto* layout = new QVBoxLayout;
+  if (event->type() == QEvent::KeyPress)
+  {
+    // Close when escape is pressed
+    if (static_cast<QKeyEvent*>(event)->matches(QKeySequence::Cancel))
+      hide();
+  }
+
+  return false;
+}
+
+void SkylanderPortalWindow::closeEvent(QCloseEvent* event)
+{
+  hide();
+}
+
+// UI
+void SkylanderPortalWindow::EmulatePortal(bool emulate)
+{
+  Config::SetBaseOrCurrent(Config::MAIN_EMULATE_SKYLANDER_PORTAL, emulate);
+  m_group_skylanders->setVisible(emulate);
+  m_command_buttons->setVisible(emulate);
+  m_emulating = emulate;
+}
+
+void SkylanderPortalWindow::SelectCollectionPath()
+{
+  QString dir = QDir::toNativeSeparators(DolphinFileDialog::getExistingDirectory(
+      this, tr("Select Skylander Collection"), m_collection_path));
+  if (!dir.isEmpty())
+  {
+    dir += QDir::separator();
+    m_path_edit->setText(dir);
+    m_collection_path = dir;
+  }
+  Config::SetBase(Config::MAIN_SKYLANDERS_PATH, dir.toStdString());
+
+  if (m_only_show_collection->isChecked())
+    RefreshList();
+}
+
+void SkylanderPortalWindow::LoadSelected()
+{
+  if (!m_emulating)
+    return;
+
+  u8 slot = GetCurrentSlot();
```

Comment:
All variables that don't mutate should be set as const.

`const u8 slot = GetCurrentSlot()`